### ISSUE TITLE
Unexpected results with mis-matched table boundaries

### ISSUE THREAD

**Issue by bkeepers** [asciidoctor/asciidoctor#923]
``` asciidoc
.Table with mismatched boundaries
|====
| Header 1 | Header 2
| Value 1  | Value 2
|===

Everything between the mismatched table boundary and the next table will disappear.

.Another Table
|====
| Header 1 | Header 2
| Value 1  | Value 2
|====

And now the rest of the doc is messed up.
```

It does give this warning, about the last line being properly formatted:

```
asciidoctor: ERROR: test.asciidoc: line 15: table missing leading separator, recovering automatically
```

I don't know what the proper way to handle this should be, but discarding the content between the tables is probably not it.

**Comment by mojavelinux** [comment]
I believe this is a duplicate of #1133 and will be addressed when #2612 is merged. An error will be reported as follows:

```
asciidoctor: ERROR: test.adoc: line 2: unterminated table block
```

**Comment by mojavelinux** [comment]
Asciidoctor now logs a warning when this happens.

**Comment by mojavelinux** [comment]
Resolved in 1631e4ffe725a79733e97948c75253fecd80bd56.
